code-name-taran/src/com/spaghettisoft/component/sixtysyx/Player.java

```diff
@@ -0,0 +1,292 @@
+package com.spaghettisoft.component.sixtysyx;
+
+import com.spaghettisoft.component.Component;
+import com.spaghettisoft.component.sixtysyx.resourses.PlayingCard;
+import com.spaghettisoft.component.sixtysyx.resourses.constants.Colour;
+import com.spaghettisoft.component.sixtysyx.resourses.constants.Rank;
+
+public class Player implements Component
+{
+	private String name;
+	private PlayingCard[] hand;
+	private int wonCardsValue;
+	public static final int PLAYER_HAND_START_SIZE = 6;
+	private int currentHandSize = PLAYER_HAND_START_SIZE;
+
+	public Player(String name)
+	{
+		this.name = name;
+		hand = new PlayingCard[PLAYER_HAND_START_SIZE];
+		wonCardsValue = 0;
+	}
+	
+	public int getCurrentHandSize()
+	{
+		return currentHandSize;
+	}
+
+	@Override
+	public void show()
+	{
+		int index = 0;
+		while (index < hand.length)
+		{
+			if (hand[index] == null)
+			{
+				index--;
+				break;
+			}
+			System.out.print((index + 1) + ".");
+			hand[index].show();
+			index++;
+		}
+
+		if (marriageInHand())
+		{
+			index++;
+			System.out.println(index + ".Announce Marriage");
+		}
+
+		if (isSwitchOptionEnabled())
+		{
+			index++;
+			System.out.println(index + ".Get Power Card!");
+		}
+
+		System.out.println();
+	}
+
+	private boolean marriageInHand()

```

Comment:
@Kristiyan-Ivanov 
The convention for naming boolean methods is to ask questions "isMarriageInHand" is better name ^_^